src/App.tsx

```diff
@@ -1,14 +1,67 @@
 /* eslint-disable max-len */
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { FilterInstructions } from './types/FilterInstructions';
+import { TodosType } from './types/TodosType';
+import { TodoModal } from './components/TodoModal';
+
+const getPreparedTodos = (
+  todosList: Todo[],
+  filterInstruction: FilterInstructions,
+) => {
+  return todosList.filter(todo => {
+    const containsSubstringInName = todo.title
+      .toLowerCase()
+      .includes(filterInstruction.todoName.toLowerCase());
+
+    if (filterInstruction.todosType === TodosType.Active) {
+      return !todo.completed && containsSubstringInName;
+    }
+
+    if (filterInstruction.todosType === TodosType.Completed) {
+      return todo.completed && containsSubstringInName;
+    }
+
+    return containsSubstringInName;
+  });
+};
 
 export const App: React.FC = () => {
+  const [todoList, setTodoList] = useState<Todo[]>([]);
+  const [chosenTodo, setchosenTodo] = useState<Todo | null>(null);
```

Comment:
Please avoid initializing your state with props. It may cause bugs in your application. Here are some articles where you can find more information about this issue: https://stackoverflow.com/a/50403930, https://reactjs.org/docs/thinking-in-react.html#step-3-identify-the-minimal-but-complete-representation-of-ui-state